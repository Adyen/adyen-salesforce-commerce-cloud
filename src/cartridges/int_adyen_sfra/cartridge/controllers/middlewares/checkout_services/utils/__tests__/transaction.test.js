jest.mock('../../helpers/index', () => ({
  checkForErrors: jest.fn(),
}));

const adyenHelpers = require('*/cartridge/scripts/checkout/adyenHelpers');
const COHelpers = require('*/cartridge/scripts/checkout/checkoutHelpers');
const basketCalculationHelpers = require('*/cartridge/scripts/helpers/basketCalculationHelpers');
const handleTransaction = require('../transaction');
const { checkForErrors } = require('../../helpers/index');

let res;
let req;
let emit;
beforeEach(() => {
  jest.clearAllMocks();
  res = { json: jest.fn() };
  req = {};
  emit = jest.fn();
});

describe('Transaction', () => {
  it('should fail if current basket has errors', () => {
    checkForErrors.mockReturnValue(true);
    const isSuccessful = handleTransaction({}, { res, req }, emit);
    expect(isSuccessful).toBeFalsy();
  });
  it('should return json with error details when payment transaction calculation fails', () => {
    checkForErrors.mockReturnValue(false);
    adyenHelpers.validatePayment.mockReturnValue({ error: false });
    COHelpers.calculatePaymentTransaction.mockReturnValue({ error: true });
    const isSuccessful = handleTransaction({}, { res, req }, emit);
    expect(res.json.mock.calls).toMatchSnapshot();
    expect(isSuccessful).toBeFalsy();
  });
  it('should succeed when payment validation and transaction calculation are successful', () => {
    checkForErrors.mockReturnValue(false);
    adyenHelpers.validatePayment.mockReturnValue({ error: false });
    COHelpers.calculatePaymentTransaction.mockReturnValue({ error: false });
    const isSuccessful = handleTransaction(
      'mockedCurrentBasket',
      { res, req },
      emit,
    );
    expect(isSuccessful).toBeTruthy();
    expect(basketCalculationHelpers.calculateTotals).toBeCalledWith(
      'mockedCurrentBasket',
    );
  });
});
