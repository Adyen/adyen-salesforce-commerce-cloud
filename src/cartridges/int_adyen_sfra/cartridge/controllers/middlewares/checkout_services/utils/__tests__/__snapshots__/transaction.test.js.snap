// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction should return json with error details when payment transaction calculation fails 1`] = `
Array [
  Array [
    Object {
      "error": true,
      "errorMessage": "mocked_error.technical",
    },
  ],
]
`;
